render/tests/test-updateElement.js

```diff
@@ -257,6 +257,79 @@ o.spec("updateElement", function() {
 
 		}
 	})
+	o("use style property setter only when cameCase keys", function() {
+		var spySetProperty = o.spy()
+		var spyRemoveProperty = o.spy()
+		var spyDashed1 = o.spy()
+		var spyDashed2 = o.spy()
+		var spyDashed3 = o.spy()
+		var spyCamelCase1 = o.spy()
+		var spyCamelCase2 = o.spy()
+
+		var f = $window.document.createElement
+		$window.document.createElement = function(tag, is){
+			var el = f(tag, is)
+
+			var style = {}
+			Object.defineProperties(style, {
+				setProperty: {value: spySetProperty},
+				removeProperty: {value: spyRemoveProperty},
+				/* eslint-disable accessor-pairs */
+				"background-color": {set: spyDashed1},
+				"-webkit-border-radius": {set: spyDashed2},
+				"--foo": {set: spyDashed3},
+				backgroundColor: {set: spyCamelCase1},
+				color: {set: spyCamelCase2}
+				/* eslint-enable accessor-pairs */
+			})
+
+			Object.defineProperty(el, "style", {value: style})
+			return el
+		}
+
+		// sets dashed properties
+		render(root, m("a", {
+			style: {
+				"background-color": "red",
+				"-webkit-border-radius": "10px",
+				"--foo": "bar"
+			}
+		}))
+
+		// sets camelCase properties and removes dashed properties
+		render(root, m("a", {
+			style: {
+				backgroundColor: "green",
+				color: "red",
+			}
+		}))
+
```

Comment:
Please test the spies here as well. We need to assert the right setter is called.